storage/country.cpp

```diff
@@ -18,119 +18,156 @@ namespace storage
     }
+    return &m_countries.AddAtDepth(depth, country);
+  }
 
-    uint32_t mwmChildCounter = 0;
-    size_t mwmChildSize = 0;
-    json_t * children = json_object_get(j, "g");
-    if (children)
-    {
-      TMwmSubtreeAttrs childAttr = LoadGroupSingleMwmsImpl(depth + 1, children, id, toDo);
-      mwmChildCounter = childAttr.first;
-      mwmChildSize = childAttr.second;
-    }
-    else
-    {
-      mwmChildCounter = 1; // It's a leaf. Any leaf contains one mwm.
-      mwmChildSize = nodeSize;
-    }
-    mwmCounter += mwmChildCounter;
-    mwmSize += mwmChildSize;
+  void InsertOldMwmMapping(TCountryId const & newId, TCountryId const & oldId) override
+  {
+    m_idsMapping[oldId].insert(newId);
+  }
 
-    if (addedNode != nullptr)
-      addedNode->SetSubtreeAttrs(mwmChildCounter, mwmChildSize);
+  void InsertAffiliation(TCountryId const & countryId, string const & affilation) override
+  {
+    m_affiliations.insert(make_pair(countryId, affilation));
   }
-  return make_pair(mwmCounter, mwmSize);
-}
 
-template <class ToDo>
-void LoadGroupTwoComponentMwmsImpl(int depth, json_t * group, TCountryId const & parent, ToDo & toDo)
+  TMappingOldMwm GetMapping() const override { return m_idsMapping; }
+};
+
+class StoreFile2InfoSingleMwms : public StoreSingleMwmInterface
 {
-  // @TODO(bykoianko) After we stop supporting two component mwms (with routing files)
-  // remove code below.
-  size_t const groupListSize = json_array_size(group);
-  for (size_t i = 0; i < groupListSize; ++i)
+  TMappingOldMwm m_idsMapping;
+  map<string, CountryInfo> & m_file2info;
+
+public:
+  StoreFile2InfoSingleMwms(map<string, CountryInfo> & file2info) : m_file2info(file2info) {}
+
+  Country * InsertToCountryTree(TCountryId const & id, uint32_t /* mapSize */, int /* depth */,
+                                TCountryId const & /* parent */) override
+  {
+    CountryInfo info(id);
+    m_file2info[id] = move(info);
+    return nullptr;
+  }
+
+  void InsertOldMwmMapping(TCountryId const & /* newId */, TCountryId const & /* oldId */) override
+  {
+  }
+  void InsertAffiliation(TCountryId const & /* countryId */,
+                         string const & /* affilation */) override
+  {
+  }
+  TMappingOldMwm GetMapping() const override
   {
-    json_t * j = json_array_get(group, i);
+    ASSERT(false, ());
+    return map<TCountryId, TCountriesSet>();
+  }
+};
+}  // namespace
 
-    char const * file = json_string_value(json_object_get(j, "f"));
-    // if file is empty, it's the same as the name
-    if (!file)
+TMwmSubtreeAttrs LoadGroupSingleMwmsImpl(int depth, json_t * node, TCountryId const & parent,
+                                         StoreSingleMwmInterface & store)
+{
+  uint32_t mwmCounter = 0;
+  size_t mwmSize = 0;
+
+  TCountryId id;
+  my::FromJSONObject(node, "id", id);
+
+  // Mapping two component (big) mwms to one componenst (small) ones.
+  json_t * oldIds = json_object_get(node, "old");
+  if (oldIds)
+  {
```

Comment:
Is it possible to reuse FromJSONObject here? It should work nice with vector of strings, right?